miniupnpd/miniupnpd.c

```diff
@@ -1192,8 +1203,12 @@ init(int argc, char * * argv, struct runtime_vars * v)
 	{
 		if(0 == strcmp(argv[i], "-h") || 0 == strcmp(argv[i], "--help"))
 			goto print_usage;
-		if(0 == strcmp(argv[i], "-d"))
-			debug_flag = 1;
+		else if(0 == strcmp(argv[i], "-d"))
+			foreground_flag = debug_flag = 1;
+#ifdef USE_SYSTEMD
+		else if(0 == strcmp(argv[i], "-D"))
+			foreground_flag = v->systemd_notify = 1;
```

Comment:
I guess it should be 
```suggestion
			foreground_flag = systemd_flag = 1;
```

Reply:
Yes, indeed, I guess it got out of sync as I was repeatedly refactoring this new code. Thanks!

Reply:
I'm putting a fixed cherrypick in master anyway

Reply:
I changed something else — mind taking a look at it again?